src/Files.App/Converters/StatusCenterStateToStateIconConverter.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Markup;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Shapes;
+
+namespace Files.App.Converters
+{
+	class StatusCenterStateToStateIconConverter : IValueConverter
+	{
+		public object? Convert(object value, Type targetType, object parameter, string language)
+		{
+			if (value is StatusCenterItemIconKind state)
+			{
+				var pathMarkup = state switch
+				{
+					StatusCenterItemIconKind.Copy =>       Application.Current.Resources["AppThemePathIconActionCopy"] as string,
+					StatusCenterItemIconKind.Move =>       Application.Current.Resources["AppThemePathIconActionMove"] as string,
+					StatusCenterItemIconKind.Delete =>     Application.Current.Resources["AppThemePathIconActionDelete"] as string,
+					StatusCenterItemIconKind.Recycle =>    Application.Current.Resources["AppThemePathIconActionDelete"] as string,
+					StatusCenterItemIconKind.Extract =>    Application.Current.Resources["AppThemePathIconActionArchive"] as string,
+					StatusCenterItemIconKind.Compress =>   Application.Current.Resources["AppThemePathIconActionArchive"] as string,
+					StatusCenterItemIconKind.Successful => Application.Current.Resources["AppThemePathIconActionSuccess"] as string,
+					StatusCenterItemIconKind.Error =>      Application.Current.Resources["AppThemePathIconActionInfo"] as string,
+					_ => ""
+				};
+
+				string xaml = @$"<Path xmlns=""http://schemas.microsoft.com/winfx/2006/xaml/presentation""><Path.Data>{pathMarkup}</Path.Data></Path>";
+
+				if (XamlReader.Load(xaml) is not Path path)
+					return null;
+
+				// Initialize a new instance
+				Geometry geometry = path.Data;
+
+				// Destroy
+				path.Data = null;
+
+				return geometry;
+			}
+
+			return null;
+		}
+
+		public object ConvertBack(object value, Type targetType, object parameter, string language)
+		{
+			throw new NotImplementedException();
+		}
+	}
```

Comment:
@yaira2 Can you test why this doesn't solve the issue where the color is not updated when the theme mode changed?

Reply:
FYI, Yair and I thought using VSManager inside of the DataTemplate would cause an issue where the ThemeResource is not updated accordingly. In fact, this approach doesn't fix that issue, though.

Reply:
The alternative approach doesn't seem to have worked. Can you try to reproduce this issue outside of Files?

Reply:
I'm sorry but, I don't think I have time to do that.

Reply:
In the meantime, let's revert back to visual states. This seems to be an issue with WinUI.

Reply:
Using VisualStateManager actually need the Template to be wraped by Control or UserControl xaml element. If either way is not a solution, I think it would be better to leave as it is.

Reply:
Should I search related issue or report this issue to the MUX team?